log-collector-script/linux/kubectl-collector.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+export LANG="C"
+export LC_ALL="C"
+readonly CURRENT_TIME=$(date +%Y-%m-%d_%H%M-%Z)
+readonly COLLECT_DIR="kubectl-output-${CURRENT_TIME}"
+mv ${COLLECT_DIR} ${COLLECT_DIR}_bk 2>/dev/null
+mkdir -p ${COLLECT_DIR}
+try() {
+  local action=$*
+  echo -n "Trying to $action... "
+}
+ok() {
+  echo
+}
+pack() {
+  try "archive gathered information"
+  tar --create --verbose --gzip --file ${PWD}/eks_kubectl_commands_"${CURRENT_TIME}".tar.gz --directory="${COLLECT_DIR}" . > /dev/null 2>&1
+  ok
+}
+finished() {
+  echo -e "\n\t Done... your kubectl command logs are located in \n \n \t ${PWD}/eks_kubectl_commands_"${CURRENT_TIME}".tar.gz \n"
+}
+try "get all Kubernetes Objects"
+kubectl get all --all-namespaces -o wide > ${COLLECT_DIR}/kubectl-get-all-${CURRENT_TIME}.log
+ok
+try "describe DaemonSet aws-node"
+kubectl describe daemonset aws-node -n kube-system > ${COLLECT_DIR}/describe-aws-node-${CURRENT_TIME}.log
+ok
+try "describe Deployment coredns"
+kubectl describe deployment coredns -n kube-system > ${COLLECT_DIR}/describe-coredns-${CURRENT_TIME}.log
+ok
+try "get all nodes"
+kubectl get nodes > ${COLLECT_DIR}/kubectl-get-nodes-${CURRENT_TIME}.log
```

Comment:
Can we make this `describe` or add `-o wide` to get some more details?

Reply:
Done. I added `-o wide`. `kubectl get all` is an expensive command for apiserver (for clusters with 100 nodes or more than 1k pods). I am afraid `get nodes -o yaml` could cause some performance issues to the control plane (or trigger CP scaling). Hence, I limited the `get nodes` to `-o wide` for now.